mmv1/third_party/tgc_next/test/assert_test_files.go

```diff
@@ -328,6 +330,45 @@ func getRoundtripConfig(t *testing.T, testName string, tfDir string, ancestryCac
 	return roundtripAssets, roundtripConfig, nil
 }
 
+// Deletes fileds from the resource data of CAI assets
+func deleteFieldsFromAsset(assets []caiasset.Asset, ignoredResourceDataFields []string) []caiasset.Asset {
+	ignoredFieldsMap := make(map[string][][]string, 0)
+	for _, ignoredField := range ignoredResourceDataFields {
+		parts := strings.Split(ignoredField, ".")
+		if len(parts) <= 1 {
+			continue
+		}
+		if parts[0] == "resource" {
+			if _, ok := ignoredFieldsMap["resource"]; !ok {
+				ignoredFieldsMap["resource"] = make([][]string, 0)
+			}
+			ignoredFieldsMap["resource"] = append(ignoredFieldsMap["resource"], parts[1:])
+		}
+	}
+
+	for _, asset := range assets {
+		if asset.Resource != nil && asset.Resource.Data != nil {
+			data := asset.Resource.Data
+			for _, ignoredFieldPathes := range ignoredFieldsMap["resource"] {
+				i := 0
+				for i < len(ignoredFieldPathes)-1 {
+					k := ignoredFieldPathes[i]
+					if v, ok := data[k]; ok {
+						data = v.(map[string]interface{})
+						i++
+					} else {
+						break
+					}
+				}
+				if i == len(ignoredFieldPathes)-1 {
+					delete(data, ignoredFieldPathes[i])
+				}
+			}
+		}
+	}
+	return assets
+}
```

Comment:
This implementation seems a bit confusing to me. Perhaps we could move L353-L365 into a helper method.

We could also add a `DeletePath` method to the `AssetResource` type and call it here.

It might also be worth refactoring so that the paths in the ignored fields map are a type that holds a slice of steps.

Reply:
Moved L353-L365 into a helper method and added a type Field.

`DeletePath` is just for tests, so I will hold it now. If necessary, we can add it later.

Reply:
Which map could be a struct here?

Reply:
I see. Do you mean ignoredFieldsMap?